python-package/pyproject.toml

```diff
@@ -63,6 +63,8 @@ disable = [
     "import-error",
     "attribute-defined-outside-init",
     "import-outside-toplevel",
+    "too-few-public-methods",
+    "too-many-ancestors",
```

Comment:
Switching the placeholder classes in the scikit-learn-is-not-available branch of `compat.py` to be individual, differently-named, empty classes (instead of all using `object`), led to these several of these `pylint` errors, in `sklearn.py` and `dask.py`:

```text
R0901: Too many ancestors (9/7) (too-many-ancestors)
R0903: Too few public methods (0/2) (too-few-public-methods)
```

([build link](https://github.com/dmlc/xgboost/actions/runs/12036652667/job/33558368143?pr=11021))

It seems that there are already many other places in this codebase where those warnings are being suppressed with `# pylint: disable` comments. So instead of adding more such comments (some of which would have to share a line with `# type: ignore` comments for `mypy`), I'm proposing:

* just globally ignore these `pylint` warnings for the whole project
* remove any existing `# pylint: disable` comments about them

I don't feel that strongly... if you'd prefer to keep suppressing individual cases of these, please let me know and I'll happily switch back to `#pylint: disable` comments.

Reply:
Looks good to me.

@RAMitchell find the pylint checks helpful. I myself prefer mypy checks and think the pylint is not particularly suitable for ML libraries like XGBoost. In general, I don't have a strong opinion about these "structural" or naming warnings and care mostly about warnings like unused imports or use before definition.